src/ol_infrastructure/applications/mitxonline/__main__.py

```diff
@@ -0,0 +1,164 @@
+                        "s3:PutObject",
+                        "s3:PutObjectAcl",
+                        "s3:GetObject*",
+                        "s3:DeleteObject*",
+                    ],
+                    "Resource": [
+                        f"arn:aws:s3:::{mitxonline_bucket_name}",
+                        f"arn:aws:s3:::{mitxonline_bucket_name}/*",
+                    ],
+                },
+            ],
+        },
+        stringify=True,
+        parliament_config={
+            "PERMISSIONS_MANAGEMENT_ACTIONS": {
+                "ignore_locations": [{"actions": ["s3:putobjectacl"]}]
+            }
+        },
+    ),
+)
+
+mitxonline_vault_backend_role = vault.aws.SecretBackendRole(
+    f"mitxonline-app-{stack_info.env_suffix}",
+    name=f"mitxonline-{stack_info.env_suffix}",
+    backend="aws-mitx",
+    credential_type="iam_user",
+    policy_arns=[mitxonline_iam_policy.arn],
+)
+
+# Create RDS instance
+mitxonline_db_security_group = ec2.SecurityGroup(
+    f"mitxonline-db-access-{stack_info.env_suffix}",
+    description=f"Access control for the MITx Online App DB in {stack_info.name}",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            protocol="tcp",
+            from_port=5432,  # noqa: WPS432
+            to_port=5432,  # noqa: WPS432
+            cidr_blocks=["0.0.0.0/0"],
+            ipv6_cidr_blocks=["::/0"],
+            description="Allow access over the public internet from Heroku",
+        )
+    ],
+    egress=[
+        ec2.SecurityGroupEgressArgs(
+            from_port=0,
+            to_port=0,
+            protocol="-1",
+            cidr_blocks=["0.0.0.0/0"],
+            ipv6_cidr_blocks=["::/0"],
+        )
+    ],
+    tags=aws_config.merged_tags(
+        {"name": "mitxonline-db-access-applications-{stack_info.env_suffix}"}
+    ),
+    vpc_id=apps_vpc["id"],
+)
+
+mitxonline_db_config = OLPostgresDBConfig(
+    instance_name=f"mitxonline-db-applications-{stack_info.env_suffix}",
+    password=mitxonline_config.require("db_password"),
+    subnet_group_name=apps_vpc["rds_subnet"],
+    security_groups=[mitxonline_db_security_group],
+    tags=aws_config.tags,
+    db_name="mitxonline",
+    public_access=True,
+    **defaults(stack_info)["rds"],
+)
+mitxonline_db = OLAmazonDB(mitxonline_db_config)
+
+mitxonline_vault_backend_config = OLVaultPostgresDatabaseConfig(
+    db_name=mitxonline_db_config.db_name,
+    mount_point=f"{mitxonline_db_config.engine}-mitxonline-{stack_info.env_suffix}",  # noqa: E501
+    db_admin_username=mitxonline_db_config.username,
+    db_admin_password=mitxonline_db_config.password.get_secret_value(),
+    db_host=mitxonline_db.db_instance.address,
+)
+mitxonline_vault_backend = OLVaultDatabaseBackend(mitxonline_vault_backend_config)
+
+######################
+# Secrets Management #
+######################
+mitxonline_secrets = vault.Mount(
+    "mitxonline-vault-secrets-storage",
+    path="secret-mitxonline",
```

Comment:
This is currently being mounted via the edxapp project. Probably worth moving this to `substructure` so that we don't run into these sorts of situations. For now, since there are no secrets being populated it might make sense to delete this section.